async-nats/src/lib.rs

```diff
@@ -1057,7 +1057,15 @@ impl ServerAddr {
 
     /// Returns the host.
     pub fn host(&self) -> &str {
-        self.0.host_str().unwrap()
+        match self.0.host() {
```

Comment:
I think `if` with `else` would be cleaner here.

Reply:
I'm not sure I follow, there are multiple cases we are matching here. What would the `if` / `else` look like?